helm-chart/make-values-tmp.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+
+import argparse
+import secrets
+import sys
+
+import ruamel.yaml
+
+yaml = ruamel.yaml.YAML()
+yaml.preserve_quotes = True
+
+
+def main():
+    """Create a values file for a secondary notebooks service helm deployment
+    (inluding a secondary Jupyterhub) which handles temporary sessions for
+    logged-out users."""
+
+    argparser = argparse.ArgumentParser(description=main.__doc__)
+
+    argparser.add_argument("--values-file", help="Input values file", required=True)
+    argparser.add_argument(
+        "--renku-namespace", help="Namespace for the Renku deployment", required=True
+    )
+    argparser.add_argument(
+        "--tmp-namespace",
+        help="Namespace for the tmp sessions deployment",
+        required=True,
+    )
+    argparser.add_argument(
+        "--output", "-o", help="Output file", default="notebooks-tmp-values.yaml"
+    )
+    args = argparser.parse_args()
+
+    with open(args.values_file) as f:
+        values = yaml.load(f.read())
+
+    # Copy over all the necessary stuff from the original values file
+    new_values = values["notebooks"]
+
+    new_values["global"] = {
+        "renku": {"domain": values["global"]["renku"]["domain"]},
+        "useHTTPS": values["global"]["useHTTPS"],
+    }
+
+    hub_section = new_values["jupyterhub"]["hub"]
+    hub_section["cookie_secret"] = secrets.token_hex(32)
+    hub_section["baseUrl"] = "{}-tmp/".format(hub_section["baseUrl"].rstrip("/"))
+    hub_section["db"]["url"] = (
+        hub_section["db"]["url"]
+        .replace("jupyterhub", "jupyterhub-tmp")
+        .replace("postgresql:", "postgresql.{}.svc:".format(args.renku_namespace))
+    )
+    hub_section["extraEnv"] = [
+        env
+        for env in hub_section["extraEnv"]
+        if env["name"] not in ["PGPASSWORD", "JUPYTERHUB_AUTHENTICATOR"]
+    ]
+    hub_section["extraEnv"].append(
+        {
+            "name": "PGPASSWORD",
+            "valueFrom": {
+                "secretKeyRef": {
+                    "name": "renku-jupyterhub-tmp-postgres",
+                    "key": "jupyterhub-tmp-postgres-password",
+                }
+            },
+        }
+    )
+    hub_section["extraEnv"].append({"name": "JUPYTERHUB_AUTHENTICATOR", "value": "tmp"})
+    hub_section["services"]["notebooks"]["url"] = "http://{}-renku-notebooks".format(
```

Comment:
this produces an url like this: `http://<namespace>-tmp-renku-notebooks`
but the correct links seems to be `http://<namespace>-renku-notebooks-tmp`

Reply:
Thanks for spotting this. It's a shortcoming of helm that there is no way to refer to the release name in a values.yaml file, so this was a hopeful guess at best. I've replaced it by setting the name of that service to `notebooks-tmp` always.